# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1533151966 14400
#      Wed Aug 01 15:32:46 2018 -0400
# Node ID e3c612c0b7445aca1c1e742002ef4eaea98d16a6
# Parent  a3448e97ccc8985e23a4f6504b75dba8fe1bf1cf
track column count in one more place in lexer

* lex.ll ("'"): Set m_current_input_column in all code branches.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -1423,30 +1423,36 @@ ANY_INCLUDING_NL (.|{NL})
                 if (tok == '[' || tok == '{'
                     || curr_lexer->previous_token_is_binop ()
                     || curr_lexer->previous_token_is_keyword ())
                   {
                     curr_lexer->m_current_input_column++;
                     curr_lexer->begin_string (SQ_STRING_START);
                   }
                 else
-                  return curr_lexer->count_token (HERMITIAN);
+                  {
+                    curr_lexer->m_current_input_column++;
+                    return curr_lexer->count_token (HERMITIAN);
+                  }
               }
           }
         else
           {
             if (! tok || tok == '[' || tok == '{' || tok == '('
                 || curr_lexer->previous_token_is_binop ()
                 || curr_lexer->previous_token_is_keyword ())
               {
                 curr_lexer->m_current_input_column++;
                 curr_lexer->begin_string (SQ_STRING_START);
               }
             else
-              return curr_lexer->count_token (HERMITIAN);
+              {
+                curr_lexer->m_current_input_column++;
+                return curr_lexer->count_token (HERMITIAN);
+              }
           }
       }
   }
 
 %{
 // Double quotes always begin strings.
 %}
 
